test/functional/aws_test.go

```diff
@@ -0,0 +1,242 @@
+package functional
+
+import (
+	goctx "context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3control"
+	"github.com/integr8ly/integreatly-operator/test/common"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+	"strings"
+	"testing"
+)
+
+var (
+	expectedRDSResources = []*rds.DBInstance{
```

Comment:
Manual test case checks also automatic backup, can this be added?

Reply:
Left this out while investigating ways of checking for that, the best solution @aidenkeating suggested was just to check if the `PreferredBackupWindow` field is not empty, That's why theres no field for it in the `DBInstance{}` slice, as we can't predict what values are going to be present, so the approach for this is to do a nil check or something similar on the `returnedInstance.PrefferedBackupWindow`, this change will come in the next few commits :+1: